core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1094,6 +1133,11 @@ public void workerRegister(long workerId, List<String> storageTiers,
       throw new NotFoundException(ExceptionMessage.NO_WORKER_FOUND.getMessage(workerId));
     }
 
+    // If this is an excluded worker, do not register any more.
+    if (isExcludedWorker(worker)) {
+      return;
+    }
```

Comment:
Since a worker only does registration when 1) at startup, 2) explicitly told by the master to do so via worker-master heartbeat, instead of rejecting the worker's registration and resetting the decommissioned state manually by the `--clear` flag later, how about simply allowing this worker to register, because when it does, it apparently has been restarted and can be put into normal service.